scripts/gen_quick_start_module.py

```diff
@@ -1,20 +1,149 @@
 #!/usr/bin/env python3
 import json
 import os
-from typing import Dict
-BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+import argparse
+import io
+import sys
+from pathlib import Path
+from typing import Dict, Set, List, Iterable
+from enum import Enum
 
+BASE_DIR =  Path(__file__).parent.parent
 
+class OperatingSystem(Enum):
+    LINUX: str = "linux"
+    WINDOWS: str = "windows"
+    MACOS: str = "macos"
+
+PRE_CXX11_ABI = "pre-cxx11"
+CXX11_ABI = "cxx11-abi"
+DEBUG = "debug"
+RELEASE = "release"
+DEFAULT = "default"
+ENABLE = "enable"
+DISABLE = "disable"
+
+# Mapping json to release matrix is here for now
+# TBD drive the mapping via:
+#  1. Scanning release matrix and picking 2 latest cuda versions and 1 latest rocm
+#  2. Possibility to override the scanning algorithm with arguments passed from workflow
+
+acc_arch_ver_map = {
+    "nightly": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        },
+    "release": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        }
+    }
+
+LIBTORCH_DWNL_INSTR = {
+        PRE_CXX11_ABI: "Download here (Pre-cxx11 ABI):",
+        CXX11_ABI: "Download here (cxx11 ABI):",
+        RELEASE: "Download here (Release version):",
+        DEBUG: "Download here (Debug version):",
+    }
+
+def load_json_from_basedir(filename: str):
+    try:
+        with open(BASE_DIR / filename) as fp:
+            return json.load(fp)
+    except FileNotFoundError as e:
+        raise ImportError(f"File {filname} not found error: {e.strerror}") from e
+    except json.JSONDecodeError as e:
+        raise ImportError(f"Invalid JSON {filname} error: {e.strerror}") from e
```

Comment:
```suggestion
        raise ImportError(f"Invalid JSON {filename} error: {e.strerror}") from e
```

Reply:
Also, are you sure that `strerror` property exists in `JSONDecodeError`? Because I can't find it:
```
>>> import json
>>> dir(json.JSONDecodeError)
['__cause__', '__class__', '__context__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__suppress_context__', '__traceback__', '__weakref__', 'args', 'with_traceback']
```